### ISSUE TITLE
[com_modules] Joomla Truncating content

### ISSUE THREAD

**Issue by Ooops-404** [joomla/joomla-cms#20724]
### What needs to be fixed
Joomla modules truncate text longer than 65,535 characters.

### Why this should be fixed
Content like the website TOS, Privacy, and so on, are not an "article" and I want to avoid users that have access to manage the articles to edit these. As well as making sure that no user in the front end navigates to these pages from an article (next, previous nav).

### How would you fix it
Changing the field type to Mediumtext

**Comment by Ooops-404** [comment]
I have a question. If I go ahead and change the field type on my sites, will future updates revert my change?

**Comment by brianteeman** [comment]
If you have content that is so long then it really should be an article. As for navigation that is simply a case of good content structure

**Comment by ghost** [comment]
Changed Title to show Issue is about com_modules.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20724">issues.joomla.org/tracker/joomla-cms/20724</a>.</sub>

**Comment by Ooops-404** [comment]
@brianteeman 

Why? why have something as an article that its not an article?
TOS and so on, are really just one page, that should not be under "articles". 

> As for navigation that is simply a case of good content structure

No really.
Let's say that I have  20 categories one being for content like TOS. Having a page that shows all article accross all categories would be off the table, or all the categories of the website, and so on.

TOS is not an article, it should not be treated like one.


Besides, what's the downside to the proposed change?

**Comment by mbabker** [comment]
Modules are not a routable part of the system.  A "page" should be created in a component, that's just the way Joomla works.  Modules are generally secondary content, there are a few rare instances where you can compose a page entirely in modules (the https://www.joomla.org homepage is one of those), but even in those cases you should not have a 65,000 character module.

**Comment by Ooops-404** [comment]
@mbabker 
My point is that a TOS page is one of those occasions and that TOS's are common to be quite extensive.
Joomla doesn't need to route modules for this to be a reality as evidenced by your own example.

That said, I think that Joomla should have a native component for instances where an article is not to be a center piece of the page. Currently this can be done by installing 3rd party extensions like "blank" or Page Builders that pretty much allow you to build a page with widgets, modules, and so on. Which is basically what I want to achieve here but without having to install any 3rd party extension. 

But I guess that's another topic.


The questions remaining being;
- What the disadvantage of changing the module content field type to Mediumtext?
- If I do it to my current Joomla installs, will that be rolledback by a future joomla update?

**Comment by mbabker** [comment]
There is no disadvantage.  But that's not the system design and personally I don't think we should make these kinds of changes to encourage this type of site design.

I've done homepages built entirely with modules, the designs used for those are great use cases for something like that "blank" component.  But a module with over 65,000 characters of text really screams "this isn't right" to me.  You're doing it to avoid exposing any com_content capabilities (page navigation, content hierarchy, etc.; all for the most part hidden through the various config options) to those pages.  Sure, it might work, but it really just doesn't look right at all.

**Comment by brianteeman** [comment]
To answer your question about losing any changes you make to the core database. 
No one can ever give you a guarantee that in the future those core modifications you make will not be overwritten

**Comment by Quy** [comment]
Related #19618

**Comment by ghost** [comment]
@Quy you can set at Tracker a "Related to" for Items. Have set it.

**Comment by Ooops-404** [comment]
@mbabker 

If there's no disadvantage then it comes down personal preference. Shouldn't Joomla be as flexible as possible? 

" But a module with over 65,000 characters of text really screams "this isn't right" to me. "

No.. not really, unless there's a technical reason behind it, is there? otherwise its again, just your preference.

" You're doing it to avoid exposing any com_content capabilities (page navigation, content hierarchy, etc.; all for the most part hidden through the various config options) to those pages."

If I can hid most but not all, then its the same as not being able to hid none at all. Wouldn't you agree?
I mean in practical terms.


So in short, there's no disadvantage at all to allow modules to hold more content, but it is what it is, because... because nothing, because it is. 
Doesn't seem the right approach to me, but that's me.

It will force me to either install a 3rd party extensions, or have 2 modules stacked each with one part of the TOS. honestly, that's what looks bad.

**Comment by mbabker** [comment]
Sure, we can make Joomla as flexible as we want.  We can also say "this use case is not in line with the system design and we recommend using the system as designed".  Part of the reason Joomla's code is so complex is because we literally try to cater to that "Joomla be as flexible as possible" statement without any real rhyme or reason why things are the way they are (well, the reason being to give people options, but then you end up with options to toggle options and options to overload options and suddenly you have 100 options for one item).

No, there is no disadvantage to allowing modules to have 125,000 characters of custom content other than changes in the database schema and inherently the resource requirements to ensure said content can be stored efficiently and correctly.  That still doesn't mean it's the right way to go about things though.

**Comment by Ooops-404** [comment]
@mbabker 

I understand that a blank statements like (ie. make it as flexible as possible in every situation) fail to address the specificities of each situation.
Each situation is different, in this case, this change wouldn't be adding any complexity. So your own blank statement about further flexibility leads to the overloading of options do not apply here.

I don't see why this is even an issue, even Joomla's own website resorts to pages with no articles in them, this alone should  be enough for everyone at Jooma to think "hey.. if we need it for our website, we know every bit of Joomla, then, chances are that this is actually a feature lacking in Joomla".

**Comment by mbabker** [comment]
The difference is Joomla's homepage isn't composed of a single module with 65,000 characters of text; it's actually several individual custom HTML modules, feed modules, and article modules.

The specific use case you're describing here is "I want to build a page with content similar to how I would put data into com_content but I don't want to use com_content".  That's not a use case I think we should be supporting.

**Comment by Ooops-404** [comment]
@mbabker 

I figured as much  and yet, I couldn't do what Joomla team did with their  homepage as Joomla doesn't have a page type for even that.  
There's no empty component in a fresh installation of joolma.

![image](https://user-images.githubusercontent.com/9021875/41418894-add6f46c-6fe8-11e8-85ad-824a386fb8c6.png)


I'm not using com_content in my user case, that's the whole problem. Nor is Joomla homepage, so the user case should be supporter for exactly that reason.

But this is a secondary question to the main question, which was; 
What's the disadvantage of changing the module content field type to Mediumtext?

As per your answer, the TLDR is none.

**Comment by brianteeman** [comment]
Sorry but I am going to close this. Having a module with more than 65k characters in it is not something we should be encouraging and the core of joomla can not ever accommodate every _crazy_ idea

For reference this privacy policy (which is overly long) is only 17,200 characters (before markup) https://www.joomla.org/privacy-policy.html
